common/math/big.go

```diff
@@ -130,6 +130,34 @@ func PaddedBigBytes(bigint *big.Int, n int) []byte {
 	return ret
 }
 
+// bigEndianByteAt returns the byte at position n,
+// if bigint is considered big-endian.
+// So n==0 returns the least significant byte
+func bigEndianByteAt(bigint *big.Int, n int) byte {
+	words := bigint.Bits()
+	// Check word-bucket the byte will reside in
+	i := n / wordBytes
+	if i >= len(words) {
+		return byte(0)
+	}
+	word := words[i]
+	// Offset of the byte
+	shift := 8 * uint(n%wordBytes)
+
+	return byte(word >> shift)
+}
+
+// Byte returns the byte at position n,
+// if bigint is considered little-endian with the supplied padlength.
+// n==0 returns the most significant byte
+// bigint '5', padlength 32, n=31 => 5
+func Byte(bigint *big.Int, padlength, n int) byte {
+	if n >= padlength {
+		return byte(0)
+	}
+	return bigEndianByteAt(bigint, padlength-1-n)
+}
```

Comment:
I still think the comments are weirdly worded. big.Int is a number and has no endianness as such. The encoding of a number has endianness. IMHO it would be better to remove the "if bigint is considered..." sentences. It's sufficient to say that "n = 0 returns the MSB".